pkg/apis/triggers/v1beta1/trigger_types.go

```diff
@@ -168,6 +168,17 @@ type GitHubInterceptor struct {
 	// +listType=atomic
 	EventTypes      []string              `json:"eventTypes,omitempty"`
 	AddChangedFiles GithubAddChangedFiles `json:"addChangedFiles,omitempty"`
+	GithubOwners    GithubOwners          `json:"githubOwners,omitempty"`
+}
+
+type GithubOwners struct {
+	Enabled bool `json:"enabled,omitempty"`
+	// This param/variable is required for private repositories or when either EnableOrgMemberCheck, EnableRepoMemberCheck are set to true
+	PersonalAccessToken *SecretRef `json:"personalAccessToken,omitempty"`
+	// Set this as true to allow org members to submit or comment on PR to proceed
```

Comment:
Are these options mutually exclusive? What happens if you set orgMemberCheck to true but repoMember to false? 

An alternative could be field that takes an enum of valid org check types - this will reduce the number of fields. Something like:

```go 
type CheckType string

const(
  OrgMembers CheckType = "orgMembers"
  RepoMembers CheckType = "repoMembers"
  None CheckType = "none"
)

type GithubOwners struct {
  ....
  CheckType CheckType `json:"memberCheck,omitempty"`
}

```

Reply:
They are not mutually exclusive. Either of the orgMemberCheck or repoMemberCheck could be true or false. Check will happen based on which is set to true.

Reply:
What happens if we set both to true?

Reply:
Then orgMemberCheck takes precedence